### PR TITLE
Expand mention to suitability of the builder

### LINKED ISSUE #66: CosmWasm core - Installation - mention that optimizer isn't suitable for dev/fast iteration
The docs should mention that the optimizer Docker image isn't suitable for fast iteration since it optimizes too much.

See https://github.com/CosmWasm/docs/pull/57#discussion_r1648739061 and https://github.com/CosmWasm/optimizer/issues/152#issuecomment-2113104183

### PR DESCRIPTION
**Pull request by aumetra**
Closes #66

### CODE REVIEW

**Review by dakom — APPROVED** [review]
Nice! 

Not-a-blocker note that might be worth a follow-up PR

**SHORT VERSION:**

I'd suggest adding something along the lines of:

> For native builds, run:
> 
> 1. cd contracts/my-contract && RUSTFLAGS="-C link-arg=-s" cargo build --release --lib --target=wasm32-unknown-unknown
> 2. wasm-opt -Os --signext-lowering "target/wasm32-unknown-unknown/release/my-contract.wasm" -o "artifacts/my-contract.wasm" 

**LONG VERSION / MOTIVATION**

For local dev what many people do is avoid the docker image entirely, and rather just use native `cargo` and `wasm-opt`, like the comment and script linked earlier: https://github.com/CosmWasm/optimizer/issues/152#issuecomment-2113104183

I have another example in Taskfile syntax here: https://github.com/dakom/ibc-chat/blob/ff384141fc58018d6ca0b55bf1fbd579451c0e9a/Taskfile.yml#L72

Tbh I got more into that habit on apple silicon a long time ago where the Docker translation layer was a _massive_ slowdown... at that time it wasn't just a "nice way to save a few milliseconds", it was crucial - but since then there were arm images and now it "just works" on arm.. maybe it's less of an issue.

Many people don't understand that the docker optimizer is not necessary at all for development or optimization, it's only really _necessary_ for reproducibility - in other words, people think if they use cargo/wasm-opt directly it's "less optimized" because they didn't use the cosmwasm tool - it's actually a fairly common misconception!

**Comment by aumetra** [comment]
@dakom Added your suggestion. Also I moved everything into a collapsible because the callout got a *little* too long

File: src/pages/core/installation.mdx [hunk 1]
```diff
@@ -24,6 +24,33 @@ platform.
...
   a Docker image to build the smallest contract possible in a deterministic fashion.
+
+  <details>
+
+    <summary>Additional information about the Optimizing Compiler</summary>
+
+    Please note that this image is _not_ suitable for development or in general
+    environments where you want to iterate quickly. The builder is optimizing for
+    size, not compilation speed. If you want to slim down your contract for
+    development, you can do so by tweaking your Cargo profile.
```
Comment (dakom) [inline src/pages/core/installation.mdx]:
```suggestion
    Please note that this image is intended for reproducible production builds.
    It is _not_ optimized for development or in general environments where you 
    want to iterate quickly. The builder is optimizing for size, not compilation speed. 
    If you want to slim down your contract for development, you can do so by 
    tweaking your Cargo profile.
```

File: src/pages/core/installation.mdx [hunk 2]
```diff
@@ -24,6 +24,33 @@ platform.
...
+
+    ```toml
+    [profile.dev]
+    lto = "thin"
+    strip = true
+    ```
+
+    If you can't use the optimizing compiler, you might miss out on determinism, but you can still build your contract into a small size like so:
```
Comment (dakom) [inline src/pages/core/installation.mdx]:
```suggestion
    If you want to build with native tools, you might miss out on determinism, but you can still build your contract into a small size like so:
```

**Review by dakom — APPROVED** [review]
LGTM!

Added a couple suggestions in comments that are purely subjective nitpicks. Absolutely feel free to ignore or accept at your discretion :)

**Pull request merged**